### ISSUE TITLE
Set error code and error message?

### ISSUE THREAD

**Issue by davidwu1010** [salesforce/reactive-grpc#234]
Hi,

How can I set error codes and error messages when using reactive-grpc with Reactor?
Currently, my grpc server only sends errors with Code: Unknown and empty messages.

Thank you.

**Comment by dan1els** [comment]
Well, I did it like that:

```
@Override
public Mono<ResponseMsg> throwE(Mono<RequestMsg> request) {
    return request
            .doOnNext(it -> {throw new IllegalStateException("Illegal state exception");})
            .onErrorMap(t -> Status.INVALID_ARGUMENT
                    .withDescription(t.getLocalizedMessage())
                    .augmentDescription("augmented message")
                    .withCause(t)
                    .asRuntimeException()
            );
}
```

It will return:

```
% grpcurl -plaintext localhost:8080 validation.ValidationService/Validate
ERROR:
  Code: InvalidArgument
  Message: Illegal state exception
augmented message
```

**Comment by dan1els** [comment]
Exceptions are actually handled with the next method: `com.salesforce.reactorgrpc.stub.ServerCalls#prepareError`
And unfortunately, all methods of that class are all static, so there's no way to enhance the logic in a legal way.

**Comment by rmichela** [comment]
`prepareError()` merely wraps an unhandled exception as a gRPC `StatusException`, if it isn't already a `StatusException` or `StatusRuntimeException`. 

I believe you can use [`onErrorResume()`](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#onErrorResume-java.util.function.Function-) to transform your unhandled Exception to a `StatusException` with whatever message you want.

**Comment by dan1els** [comment]
That's correct, but it's not possible to introduce some common logic for exception handling because of that.
For instance, we're using spring grpc starter, and it contains very handy `@GrpcAdvice` and `@GrpcExceptionHandler` working closer to its REST analogs.
But it can't work with this library together because this library has that static `prepareError()` which goes before stater interceptor, and it's static too, so there's no way to get them work together.

**Comment by dan1els** [comment]
Yeah, i'll try to find the time to do that

**Comment by lobanovdmitry** [comment]
@rmichela Hello, just want to ask about related issue we faced.
Reactor doesn't handle "Jvm Fatal Errors" (https://projectreactor.io/docs/core/release/reference/#_handling_exceptions_in_operators_or_functions), so, for example: if somewhere in the code such exceptions occur (for example NoSuchMethodError), we can't handle it using onError(), etc... While such errors are handled in https://github.com/salesforce/reactive-grpc/blob/master/reactor/reactor-grpc-stub/src/main/java/com/salesforce/reactorgrpc/stub/ServerCalls.java#L70. So, for such issues we see even no logging on server side(and there is not way to add it) and client just receives error with UNKNOWN status. Can you recommend how to deal with that?
